CommonMark/CommonMark.py

```diff
@@ -49,15 +49,13 @@ def prepare(block):
         block.__dict__[r] = None
     if block.is_open is not None:
         block.__dict__['open'] = block.is_open
-        del(block.is_open)
+        del block.is_open
+
     # trim empty elements...
-    for attr in dir(block):
-        if not callable(attr) and not attr.startswith("__") and \
-           attr not in ['pretty', 'is_container',
-                        'append_child', 'prepend_child', 'unlink',
-                        'insert_after', 'insert_before', 'walker']:
-            if block.__dict__[attr] in ["", [], None, {}]:
-                del(block.__dict__[attr])
+    for attr, value in list(block.__dict__.items()):
```

Comment:
Is it considered best practice to rely on using `__dict__`? I know the existing code does it, but it would be nice to completely clean this up. I'm just curious.